release/charts/charts.go

```diff
@@ -0,0 +1,61 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	ecmConfig "github.com/rancher/ecm-distro-tools/cmd/release/config"
+)
+
+// ListLifecycleStatus prints the lifecycle status of the charts
+func ListLifecycleStatus(ctx context.Context, c *ecmConfig.ChartsRelease, branch, chart string) error {
+
+	var branchArg, chartArg string = "", ""
+
+	branchArg = "--branch-version=" + branch
+	if chart != "" {
+		chartArg = "--chart=" + chart
+	}
+
+	err := executeChartsBuildScripts(c.Workspace, "lifecycle-status", branchArg, chartArg)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("generated log files for inspection at: \n%s\n", c.Workspace+"/logs/")
+	return nil
+}
+
+func executeChartsBuildScripts(chartsRepoPath string, args ...string) error {
+	// save current working dir
+	ecmWorkDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// change working dir to the charts repo
+	err = os.Chdir(chartsRepoPath)
```

Comment:
Move the if check up a line and scope lock the err value. Please do this for all calls that return a single err value.